### ISSUE TITLE
How do I use the model from federated learning to output predictive values？

### ISSUE THREAD

**Issue by skiifall** [tensorflow/federated#2677]
After federated learning, I want to input some x and use the model directly to get some predicted values. Can I use the model directly to TensorFlow in general?The current model is wrapped in an interface, and I can't seem to use it directly.

**Comment by hardik-vala** [comment]
Hi, can you use the [`predict_on_batch`](https://github.com/tensorflow/federated/blob/main/tensorflow_federated/python/learning/model.py#L162) method on the model to run inference for your case?

**Comment by skiifall** [comment]
> Hi, can you use the [`predict_on_batch`](https://github.com/tensorflow/federated/blob/main/tensorflow_federated/python/learning/model.py#L162) method on the model to run inference for your case?
I build the model using tff.learning.from_keras_model() instead of directly building tff.learning.Model using the lower-level model interface. the former may not involve predict_on_batch(). However, I only found the federated data evaluation function tff.learning.build_federated_evaluation() in the official documentation, and nothing similar to Keras' model.predict function

**Comment by skiifall** [comment]
> 您好，您可以使用[`predict_on_batch`](https://github.com/tensorflow/federated/blob/main/tensorflow_federated/python/learning/model.py#L162)模型上的方法为您的案例运行推理吗？

Hello, I have implemented federated learning according to the routine custom model and also plan to use the predict_on_batch () function to get the predictions. But I don't know where I should get variables type parameters in predict_on_batch (variables, x).

**Comment by hardik-vala** [comment]
Can you share a code snippet of what you're trying to do?

**Comment by skiifall** [comment]
> 你能分享一个你正在尝试做的代码片段吗？
Hello, I think the problem has been solved. I didn't use predict_on_batch() because I was trying to find the parameters required by the function but failed. My solution is as follows, 
1.I redefine a keras.model that is the same as the federated learning definition
2.I set weights into the new independent model bying using the assign_weights_to() in tff. learning. ModelWeights weights.
3.At this point I can probably use the new Keras.model to get the predictions.
